zerver/models.py

```diff
@@ -1011,6 +1011,11 @@ def get_context_for_message(message):
 
 post_save.connect(flush_message, sender=Message)
 
+class Reaction(ModelReprMixin, models.Model):
+    sender = models.ForeignKey(UserProfile) # type: UserProfile
+    message = models.ForeignKey(Message) # type: Message
+    emoji = models.ForeignKey(RealmEmoji) # type: RealmEmoji
```

Comment:
change this to a `TextField()` (matching RealmEmoji) for the emoji name, and update the migration.